### DISCUSSION TITLE
Possible issue with the verification of cleartext signed message

### DISCUSSION THREAD

**Discussion by BRULE-Florian** [pgpainless/pgpainless#217] [Q&A] [answered]
Hi,

When I call `PGPainless.verifyCleartextSignedMessage()` with an `Inputstream` that contains a **too long** _cleartext_ signed message (**maximum 8191 characters allowed**), it doesn't work correctly.
It throws `java.io.IOException: Resetting to invalid mark`.
Is there a **solution** to verify my signed message **or** is there really a **bug** ?

Here is the code that causes the exception:
```
String signedMessage = "...";
BufferedInputStream signedMessageStream = new BufferedInputStream(
    new ByteArrayInputStream(signedMessage.getBytes(StandardCharsets.UTF_8)));

ConsumerOptions options = new ConsumerOptions()
    .addVerificationCert(publicKey)
    .addVerificationOfDetachedSignature(signature);

DecryptionStream verificationStream = PGPainless.verifyCleartextSignedMessage()
    .onInputStream(signedMessageStream)
    .withStrategy(new InMemoryMultiPassStrategy())
    .withOptions(options)
    .getVerificationStream();
```
_The exception is thrown in the method `getVerificationStream()`._

I am at your disposal if you have any questions or need more details.

Kind regards,
Florian

**Comment by vanitasvitae** [comment]
This is most likely a bug. I will investigate.

**Reply by BRULE-Florian** [reply]
Thanks.
If you want help, I'm here :wink: .

**Comment by vanitasvitae** [comment, marked as answer]
I just released PGPainless 1.0.0-rc5 which should fix this issue.
